core/serialize/serialize.cc

```diff
@@ -1125,6 +1125,31 @@ void Serializer::loadGlobalState(GlobalState &gs, const uint8_t *const symbolTab
     }
 }
 
+void Serializer::initForSlowPath(GlobalState &result, const GlobalState &gs, const uint8_t *const symbolTableData) {
```

Comment:
This is a weird function that I couldn't really think of a better home for: it needs to selectively copy parts of the global state and then initialize the symbol table with the payload.

Reply:
I think that we should build this on top of `copyForIndex`.

I think we should make a new `GlobalState::copyForSlowPath(gs)` method that

- calls `copyForIndex(gs)`
- copies the string- and name-related tables from `gs`

and then I think we should make a `pipeline::copyForSlowPath(gs)` that

- calls `result = gs.copyForSlowPath(gs)`
- calls `unpickleSymbolTable(result)`

I do not think that a `copyGlobalState` function should live in `serialize.cc`, and also I think we should avoid new methods in `payload` in favor of `pipeline`.

Let me know if that sounds good to you. If it's unclear, I think I might be able to show what I'm thinking better by committing to your branch.

Reply:
I think that makes sense. I'll implement what I think you've described and ping for a re-review.

Reply:
I think we can't build it on top of `copyForIndex` as we shouldn't call `initEmpty` when we're creating this copy: we're going to initialize the symbol table with the payload, which includes what `initEmpty` would have added.